### ISSUE TITLE
no-duplicate-variable does not handle destructuring

### ISSUE THREAD

**Issue by leeavital** [palantir/tslint#425]
```typescript
var a : number = 1;
var [a, b] : number[] = [2,3];
````

Should warn about duplicating `a`.

**Comment by adidahiya** [comment]
I have code for this somewhere in my fork of tslint. I'll investigate its status.

**Comment by adidahiya** [comment]
yep, I've added destructuring support @ this branch: https://github.com/adidahiya/tslint/tree/destructuring-no-duplicate-variable

but it depends on a currently open PR #409

**Comment by adidahiya** [comment]
sorry, will close when #452 merges
